src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -212,7 +212,6 @@ public void setValidateOnlyOverlapping(boolean validateOnlyOverlapping) {
     @Override
     public void beginTree(DetailAST rootAST) {
         frames = new HashMap<>();
-        current.clear();
```

Comment:
`clear` can't be removed here.
`current` may be completely empty after a file's good run, but if an exception happens and `haltOnException` is off, it will keep contents from previous file's run.
See https://github.com/checkstyle/checkstyle/issues/2939#issuecomment-304343092 as an example.